src/main/java/org/mapstruct/intellij/inspection/TargetPropertyMappedMoreThanOnceInspection.java

```diff
@@ -0,0 +1,210 @@
+                if ( MAPPING_ANNOTATION_FQN.equals( qualifiedName ) ) {
+                    handleMappingAnnotation( psiAnnotation, problemMap );
+                }
+                else if (MAPPINGS_ANNOTATION_FQN.equals( qualifiedName )) {
+                    extractMappingAnnotationsFromMappings( psiAnnotation )
+                            .forEach( a -> handleMappingAnnotation( a, problemMap ) );
+                }
+                else {
+                    // Handle annotations containing at least one Mapping annotation
+                    handleAnnotationWithMappingAnnotation( psiAnnotation, problemMap );
+                }
+            }
+            QuickFixFactory quickFixFactory = QuickFixFactory.getInstance();
+            for (Map.Entry<String, List<PsiElement>> problem : problemMap.entrySet()) {
+                List<PsiElement> problemElements = problem.getValue();
+                if (problemElements.size() > 1) {
+                    for (PsiElement problemElement : problemElements) {
+                        LocalQuickFix[] quickFixes = getLocalQuickFixes( problemElement, quickFixFactory );
+                        holder.registerProblem( problemElement,
+                               MapStructBundle.message( "inspection.target.property.mapped.more.than.once",
+                                       problem.getKey() ), quickFixes );
+                    }
+                }
+            }
+        }
+
+        private static @NotNull  LocalQuickFix[] getLocalQuickFixes(PsiElement problemElement,
+                                                                       QuickFixFactory quickFixFactory) {
+            List<LocalQuickFix> quickFixes = new ArrayList<>(2);
+            if (problemElement instanceof PsiAnnotation) {
+                quickFixes.add( getDeleteFix( problemElement, quickFixFactory ) );
+            }
+            else if (problemElement instanceof PsiAnnotationMemberValue) {
+                Optional.ofNullable( problemElement.getParent() ).map( PsiElement::getParent )
+                        .map( PsiElement::getParent ).filter( PsiAnnotation.class::isInstance )
+                        .ifPresent( annotation -> quickFixes.add(
+                                getDeleteFix( annotation, quickFixFactory ) ) );
+                quickFixes.add( new ChangeTargetQuickFix( (PsiAnnotationMemberValue) problemElement) );
+            }
+            return quickFixes.toArray( new LocalQuickFix[]{} );
+        }
+
+        private static @NotNull LocalQuickFixAndIntentionActionOnPsiElement getDeleteFix(
+                PsiElement problemElement, QuickFixFactory quickFixFactory) {
+            String qualifiedName = Objects.requireNonNullElse(
+                    ( (PsiAnnotation) problemElement).getQualifiedName(), "unknown" );
+            return quickFixFactory.createDeleteFix( problemElement,
+                    MapStructBundle.message( "intention.remove.annotation", qualifiedName ) );
```

Comment:
Would it be possible to use the simple name with a leading `@` so that the message looks like `Remove @Mapping annotation`?

![example](https://github.com/user-attachments/assets/32897f50-e7c8-407a-ab24-3e8b56768bff)

Reply:
Now only the simple name is shown.  I didn't add the leading `@` as IntelliJ itself also doesn't show the leading `@`  for annotation related quick fixes.